hardware_interface/include/joint_limits_interface/joint_limits_interface.hpp

```diff
@@ -45,33 +46,33 @@ class JointLimitHandle
 {
 public:
   JointLimitHandle()
-  : prev_pos_(std::numeric_limits<double>::quiet_NaN()),
-    prev_vel_(0.0),
-    jposh_("position"),
+  : jposh_("position"),
     jvelh_("velocity"),
-    jcmdh_("position_command")
+    jcmdh_("position_command"),
+    prev_pos_(std::numeric_limits<double>::quiet_NaN()),
+    prev_vel_(0.0)
   {}
 
   JointLimitHandle(
-    const hardware_interface::JointHandle & jposh,
-    const hardware_interface::JointHandle & jcmdh,
+    const hardware_interface::StateInterface & jposh,
+    hardware_interface::CommandInterface && jcmdh,
     const JointLimits & limits)
   : jposh_(jposh),
     jvelh_("velocity"),
-    jcmdh_(jcmdh),
+    jcmdh_(std::move(jcmdh)),
     limits_(limits),
     prev_pos_(std::numeric_limits<double>::quiet_NaN()),
     prev_vel_(0.0)
   {}
 
   JointLimitHandle(
-    const hardware_interface::JointHandle & jposh,
-    const hardware_interface::JointHandle & jvelh,
-    const hardware_interface::JointHandle & jcmdh,
+    const hardware_interface::StateInterface & jposh,
+    const hardware_interface::StateInterface & jvelh,
+    hardware_interface::CommandInterface && jcmdh,
```

Comment:
We should probably use different names to correspond to the new nomenclature:
```suggestion
    const hardware_interface::StateInterface & joint_pos_interface,
    const hardware_interface::StateInterface & joint_vel_interface,
    hardware_interface::CommandInterface && joint_cmd_interface,
```

This should be probably done in a follow-up PR.